trade-finance-manager-api/src/v1/services/entra-id.service.ts

```diff
@@ -24,37 +24,72 @@ type GetAccessTokenAndEntraIdUserByAuthCodeParams = {
 };
 
 type GetEntraIdUserByAuthCodeResponse = {
-  entraIdUser: EntraIdUser;
+  idTokenClaims: EntraIdUser;
 };
 
 export class EntraIdService {
-  private readonly msalAppConfig: MsalAppConfig;
-
+  private readonly msalConfig: MsalAppConfig;
   private readonly cryptoProvider = new CryptoProvider();
   private readonly entraIdConfig: EntraIdConfig;
   private readonly entraIdApi: EntraIdApi;
+
   constructor({ entraIdConfig, entraIdApi }: { entraIdConfig: EntraIdConfig; entraIdApi: EntraIdApi }) {
     this.entraIdConfig = entraIdConfig;
     this.entraIdApi = entraIdApi;
-    this.msalAppConfig = entraIdConfig.msalAppConfig;
+    this.msalConfig = entraIdConfig.msalConfig;
   }
 
   /**
-   * Used as part of the SSO process
+   * Retrieves an instance of the `ConfidentialClientApplication` configured with
+   * the necessary authority metadata to optimise performance. If the authority
+   * metadata is not already cached in the `msalConfig`, it fetches the metadata
+   * and stores it for future use.
    *
-   * Creates the URL to redirect the user to Entra to login
+   * @returns {Promise<ConfidentialClientApplication>} A promise that resolves to
+   * an instance of `ConfidentialClientApplication` configured with the cached
+   * authority metadata.
    *
-   * The redirect URI is retrieved from Entra later in the process, following a successful login, to allow
-   * the user to be redirected to the original page they were visiting.
+   * @remarks
+   * This method improves performance by avoiding repeated metadata fetches on
+   * every request. For more details, refer to the MSAL Node performance documentation:
+   * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/performance.md
+   *
+   * @throws {Error} If unable to get the MSAL application instance.
+   */
+  private async getMsalInstance(): Promise<ConfidentialClientApplication> {
+    try {
+      if (!this.msalConfig.auth.authorityMetadata) {
+        const metaData = await this.entraIdApi.getAuthorityMetadataUrl();
+        this.msalConfig.auth.authorityMetadata = JSON.stringify(metaData);
+      }
+
+      return new ConfidentialClientApplication(this.msalConfig);
+    } catch (error) {
+      console.error('Unable to get MSAL instance %o', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Generates an authorisation code URL for Entra ID authentication.
+   *
+   * This method creates a URL that can be used to initiate the OAuth 2.0
+   * authorization code flow with Entra ID. It includes a CSRF token in the
+   * state parameter to protect against CSRF attacks, and encodes the state
+   * as a base64 string.
+   *
+   * @param {GetAuthCodeUrlParams} params - Parameters for generating the authorization code URL.
```

Comment:
@abhi-markan shouldn't be this 
```suggestion
   * @param {GetAuthCodeUrlParams} successRedirect - Parameters for generating the authorization code URL.
```